File: attached_assets/Starter_code/Dotnet_starter_code/Insurance.Api/Services/UnderwritingService.cs
namespace Insurance.Api.Services;

public class UnderwritingService
{
    public object Evaluate(int age, string product, int priorClaims)
    {
        double risk = (age / 10.0) + (priorClaims * 5.0);
        if (string.Equals(product, "life", StringComparison.OrdinalIgnoreCase)) risk += 10;
        if (string.Equals(product, "home", StringComparison.OrdinalIgnoreCase)) risk += 5;
        var decision = risk < 10 ? "approve" : risk < 20 ? "review" : "decline";
        return new { riskScore = Math.Round(risk), decision };
    }
}

File: attached_assets/Starter_code/Java_starter_code/src/main/java/com/example/insurance/service/UnderwritingService.java
package com.example.insurance.service;

import org.springframework.stereotype.Service;

@Service
public class UnderwritingService {
    public Result evaluate(int age, String product, int priorClaims) {
        double risk = (age / 10.0) + (priorClaims * 5.0);
        if ("life".equalsIgnoreCase(product)) risk += 10;
        if ("home".equalsIgnoreCase(product)) risk += 5;
        String decision = risk < 10 ? "approve" : risk < 20 ? "review" : "decline";
        return new Result(Math.round(risk), decision);
    }

    public record Result(long riskScore, String decision) {}
}


